### PR TITLE
add CGI product page.

### PR DESCRIPTION
**Pull request by ztcrypto**
missing pieces of api usage.

### CODE REVIEW

File: src/views/CGI/CGI.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    </Page>
+  )
+}
+
+const StyledDpiIndexCalculationImage = styled.img`
+  margin-bottom: 20px;
+  width: 100%;
+`
+
+export default DpiProductPage
```
Comment (controtie) [inline src/views/CGI/CGI.tsx:260]: rename

File: src/views/CGI/CGI.tsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            The CGCI-LV therefore deploys the same theory but instead limits the
+            index to 3 ERC-20 token constituents; a gold stablecoin and 2
+            cryptoassets. For the gold basket, wrapped-DGLD (wDGLD) was chosen
+            as the ERC-20 gold stablecoin.
+            <br />
+            For the cryptoasset basket wrapped-bitcoin (wBTC) and wrapped-Ether
+            (wETH) were selected based on being the largest and most established
+            cryptoassets.
+            <h2>Objective</h2>
+            The DeFi Pulse Index is a digital asset index designed to track
```
Comment (controtie) [inline src/views/CGI/CGI.tsx:86]: Some mentions of DeFi Pulse here

File: src/views/CGI/components/MarketData.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  const priceData = new Date(chartDate)
+
+  return (
+    <div>
+      <StyledDpiIconLabel>
+        <StyledIcon src={dpiTokenIcon.src} alt={dpiTokenIcon.alt} />
+        <span>DPI</span>
+      </StyledDpiIconLabel>
+      <StyledDpiTitle>DeFi Pulse Index</StyledDpiTitle>
```
Comment (controtie) [inline src/views/CGI/components/MarketData.tsx:73]: rename to CGI

**Review by controtie — COMMENTED** [review]
looks quite good! i'll let you know when the API is up ASAP

File: src/views/CGI/CGI.tsx (hunk 2 above)
Comment (ztcrypto) [reply src/views/CGI/CGI.tsx]: fixed

File: src/views/CGI/CGI.tsx (hunk 1 above)
Comment (ztcrypto) [reply src/views/CGI/CGI.tsx]: fixed

File: src/utils/tokensetsApi.ts [hunk 4]
```diff
@@ -47,3 +47,58 @@ export const fetchSetComponents = (set: string) => {
...
     .catch((error) => console.log(error))
 }
+
+export const fetchHistoricalTokenMarketData = (
+  id: string,
+  baseCurrency = 'usd'
+) => {
+  const baseURL = 'https://api.tokensets.com/v2/fund_historicals'
+  const requestUrl =
+    baseURL + `/${id}` + `?currency=${baseCurrency}&beta=true&interval=month`
```
Comment (controtie) [inline src/utils/tokensetsApi.ts:57]: combine baseURL + `/${id}?currency=${baseCurrency}&beta=true&interval=month`

File: src/utils/tokensetsApi.ts [hunk 5]
```diff
@@ -47,3 +47,58 @@ export const fetchSetComponents = (set: string) => {
...
+
+  return fetch(requestUrl)
+    .then((response) => response.json())
+    .then((response) => {
+      /** @notes from API docs
+       * Minutely data will be used for duration within 1 day
+       * Hourly data will be used for duration between 1 day and 90 days
+       * Daily data will be used for duration above 90 days.
+       */
```
Comment (controtie) [inline src/utils/tokensetsApi.ts:66]: Is this note necessary?

File: src/utils/tokensetsApi.ts [hunk 6]
```diff
@@ -47,3 +47,58 @@ export const fetchSetComponents = (set: string) => {
...
+      const { prices, dates } = response
+      return {
+        prices: prices.map((item: any, index: number) => [dates[index], item]),
+      }
+    })
+    .catch((error) => console.log(error))
+}
+
+export const fetchSetComponentsBeta = (set: string) => {
+  // const requestUrl = `${baseURL}/v2/portfolios/${set}`
```
Comment (controtie) [inline src/utils/tokensetsApi.ts:76]: lint

File: src/views/CGI/CGI.tsx [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+} from 'components/ProductPage'
+import { BuySellWrapper } from 'components/BuySell'
+import ExternalLink from 'components/ExternalLink'
+import MarketData from './components/MarketData'
+
+import useCgiTokenMarketData from 'hooks/useCgiTokenMarketData'
+import useCgiIndexComponents from 'hooks/useCgiIndexComponents'
+import useBalances from 'hooks/useBalances'
+
+import DpiIndexCalculationImage from 'assets/dpi-index-calculation.png'
```
Comment (controtie) [inline src/views/CGI/CGI.tsx:23]: remove

File: src/views/CGI/CGI.tsx [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+
+  const [marketCap, setMarketCap] = useState(0)
+
+  useEffect(() => {
+    fetchHistoricalTokenMarketData(coingeckoCgiId)
+      .then((response: any) => {
+        setMarketCap(response)
+      })
+      .catch((error: any) => console.log(error))
+  }, [])
```
Comment (controtie) [inline src/views/CGI/CGI.tsx:40]: remove

Comment (controtie) [reply src/views/CGI/CGI.tsx:40]: when we eventually use this in the future, it should come from a provider right?

File: src/views/CGI/CGI.tsx [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+        </ProductPageContent>
+      </Container>
+    </Page>
+  )
+}
+
+const StyledCgiIndexCalculationImage = styled.img`
+  margin-bottom: 20px;
+  width: 100%;
+`
```
Comment (controtie) [inline src/views/CGI/CGI.tsx:255]: remove

**Review by controtie — CHANGES_REQUESTED** [review]
Looks good so far. Left some clean up comments. In addition:

Remove show less button here:
image

Remove this section for now until coingecko api is set up.
image

File: src/views/CGI/CGI.tsx (hunk 8 above)
Comment (ztcrypto) [reply src/views/CGI/CGI.tsx:40]: yes

**Review by controtie — APPROVED** [review]
LGTM!

**Pull request merged**